src/ORM/Queries/SQLSelect.php

```diff
@@ -250,10 +250,17 @@ public function setLimit($limit, $offset = 0)
             throw new InvalidArgumentException("SQLSelect::setLimit() only takes positive values");
         }
 
+        if ($limit === 0) {
+            user_error(
+                "Query passed a limit of 0; this will return all records. To clear limit, set it to null",
+                E_USER_WARNING
```

Comment:
Since we're going to change this to return zero records in SS5, do you think we should mention that in the warning message?

Reply:
I'm going to replace with a deprecation notice.